page.jsx

```diff
@@ -121,14 +121,19 @@ function ComponentPage({
   swingsetOptions,
   peerComponents,
 }) {
+  const { default: defaultExport, ...namedExports } = component.exports
   return (
     <MDXRemote
       {...mdxSource}
       components={createScope(
-        { [component.data.componentName]: component.src },
+        {
+          [component.data.componentName]: defaultExport,
+          ...namedExports,
+        },
         swingsetOptions,
         peerComponents
       )}
+      scope={{ ...mdxSource.scope, ...namedExports }}
```

Comment:
we could probably clean this up and try to detect react components and only spread in `components` or `scope` depending on if it's a component or not, but this seems to do the job for now!